src/lib/Telemetry/telemetry.cpp

```diff
@@ -190,120 +190,77 @@ bool Telemetry::RXhandleUARTin(uint8_t data)
-
-        if (header->type == CRSF_FRAMETYPE_ARDUPILOT_RESP)
+        if (header->type == payloadTypes[i].type && CRSF_FRAME_SIZE(header->frame_size) <= payloadTypes[i].size)
         {
-            // reserve last slot for adrupilot custom frame with the sub type status text: this is needed to make sure the important status messages are not lost
-            if (package[CRSF_TELEMETRY_TYPE_INDEX + 1] == CRSF_AP_CUSTOM_TELEM_STATUS_TEXT)
-            {
-                targetIndex = payloadTypesCount - 1;
-            }
-            else
-            {
-                targetIndex = payloadTypesCount - 2;
-            }
-            targetFound = true;
-        }
-        else if (extHeader->orig_addr == CRSF_ADDRESS_FLIGHT_CONTROLLER)
-        {
-            targetIndex = payloadTypesCount - 2;
-            targetFound = true;
-
-            #if defined(USE_MSP_WIFI) && defined(TARGET_RX)
-                // this probably needs refactoring in the future, I think we should have this telemetry class inside the crsf module
-                if (wifi2tcp.hasClient() && (header->type == CRSF_FRAMETYPE_MSP_RESP || header->type == CRSF_FRAMETYPE_MSP_REQ)) // if we have a client we probs wanna talk to it
-                {
-                    DBGLN("Got MSP frame, forwarding to client, len: %d", currentTelemetryByte);
-                    CRSF::crsf2msp.parse(package);
-                }
-                else // if no TCP client we just want to forward MSP over the link
-            #endif
-            {
-                // larger msp resonses are sent in two chunks so special handling is needed so both get sent
-                if (header->type == CRSF_FRAMETYPE_MSP_RESP)
-                {
-                    // there is already another response stored
-                    if (payloadTypes[targetIndex].updated)
-                    {
-                        // use other slot
-                        targetIndex = payloadTypesCount - 1;
-                    }
-
-                    // if both slots are taked do not overwrite other data since the first chunk would be lost
-                    if (payloadTypes[targetIndex].updated)
-                    {
-                        targetFound = false;
-                    }
-                }
-            }
-        }
-        else
-        {
-            targetIndex = payloadTypesCount - 1;
+            targetIndex = i;
             targetFound = true;
         }
     }
-    else
+
+    // store anything else in 2-slot FIFO buffer, do not overwrite existing data. This also handles larger msp responses, which are sent in two chunks 
+    if (!targetFound)
     {
-        for (int8_t i = 0; i < payloadTypesCount - 2; i++)
+        // first try slot payloadTypesCount - 2, so that the OTA packets are transmitted in the same order as they are received
+        targetIndex = payloadTypesCount - 2;
+        targetFound = !payloadTypes[targetIndex].updated;
```

Comment:
If we decide to make it compatible to the current behaviour we can change this line to 
```cpp
targetFound = !payloadTypes[targetIndex].updated || !(header->orig_addr == CRSF_ADDRESS_FLIGHT_CONTROLLER || (header->type == CRSF_FRAMETYPE_ARDUPILOT_RESP && package[CRSF_TELEMETRY_TYPE_INDEX + 1] == CRSF_AP_CUSTOM_TELEM_STATUS_TEXT));
```

Reply:
Thanks. I've added the fix, to see if the test passes now. 

But I'm not 100% sure that this is the best way to go. It looks to me that after this fix we're back to single buffering most messages, and thus back to the old situation slowing things down artificially - the exact opposite of what this PR tries to achieve.

I'm working on the streaming PR now, but time permitting I'll run the throughput tests with this fix in place.

Reply:
Thats why I gave the options 😄, and said that personally I'd go for the simple option i.e. fix the unit test, and make it check for 2 messages.

Reply:
If we really want a special case for the ardupilot text message being only a single slot, then we should just do that and allow all others to be double slot.

Reply:
@pkendall64 I missed your hint about changing the unit tests - still was under the impression that changing the unit tests was a no go :-)

Changed back things back to the original PR code, and changed the unit test to check for 2 messages.

Reply:
I've added an additional slot for StatusText.

I've been using this PR code (without additional StatusText slot) with Ardupilot & Yaapu and status text worked. StatusText could be missed when flooding the receiver with other CRSF messages. However the original code has the same issue, as it stores StatusText in slot ```targetIndex = payloadTypesCount - 1``` which is also used for other CRSF messages.

So, I've added a separate StatusText slot, to handle this case separately and always keep the last StatusText buffered to be sent OTA.

Reply:
@schugabe Please let me know if the addition I made eases your concerns, or do you require further changes?

Reply:
The overflow was possible but yapuu tuned the sending rate to prevent issues. Did you test this PR with the yapuu script at 50hz update rate + 1:2 tlm?